iohub/ngff/nodes.py

```diff
@@ -675,22 +670,24 @@ def create_image(
             chunks = self._default_chunks(data.shape, 3)
         if check_shape:
             self._check_shape(data.shape)
-        img_arr = ImageArray(
-            self._group.array(
-                name, data, chunks=chunks, **self._storage_options
+        img_arr = ImageArray.from_zarr_array(
+            self._group.create_array(
+                name=name, shape=data.shape, dtype=data.dtype, chunks=chunks, overwrite=self._overwrite,
+                **self._create_compressor_options(chunks),
             )
         )
+        img_arr[...] = data
```

Comment:
This breakage seems unintended from upstream discussion. We can revert after https://github.com/zarr-developers/zarr-python/pull/2819 is released.